### ISSUE TITLE
checkers: boolExprSimplify false positive for floats

### ISSUE THREAD

**Issue by quasilyte** [go-critic/go-critic#848]
Triggered at https://github.com/flyingmutant/rapid/blob/master/utils.go#L29.

```
./src/github.com/flyingmutant/rapid/utils.go:29:9: boolExprSimplify: can simplify `p > 0 && p <= 1` to `p == 1`
```
Suggested replacement only makes sense for int values.
Referenced code uses floating point values.
